python/tests/operations/join.py

```diff
@@ -275,6 +275,55 @@ def test_different_names(self):
 
         self.assertTrue(np.equal(joined_labels.tolist(), ref).all())
 
+    def test_different_names_different_length(self):
+        """Test Label joining using labels with different names and different length."""
+        property_labels_1 = Labels(
+            ("structure", "prop_1"), np.vstack(2 * [np.arange(5)]).T
+        )
+        property_labels_2 = Labels(
+            ("structure", "prop_2", "prop_3"), np.vstack(3 * [np.arange(5)]).T
+        )
+
+        block_1 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_1,
+        )
+
+        block_2 = TensorBlock(
+            values=np.zeros([2, 5]),
```

Comment:
shouldn't this be `np.zeros([5,2])`?  Since these are `property_labels_2`

Reply:
Why. These are the values of the Block. It has two samples and 5 properties.